server/src/main/java/org/elasticsearch/indices/recovery/RecoveryFinalizeRecoveryRequest.java

```diff
@@ -19,30 +19,39 @@
 
 package org.elasticsearch.indices.recovery;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.index.seqno.SequenceNumbers;
 import org.elasticsearch.index.shard.ShardId;
 import org.elasticsearch.transport.TransportRequest;
 
 import java.io.IOException;
 
-public class RecoveryFinalizeRecoveryRequest extends TransportRequest {
+final class RecoveryFinalizeRecoveryRequest extends TransportRequest {
 
-    private long recoveryId;
-    private ShardId shardId;
-    private long globalCheckpoint;
+    private final long recoveryId;
+    private final ShardId shardId;
+    private final long globalCheckpoint;
+    private final long startingSeqNo;
 
-    public RecoveryFinalizeRecoveryRequest(StreamInput in) throws IOException {
+    RecoveryFinalizeRecoveryRequest(StreamInput in) throws IOException {
         super(in);
         recoveryId = in.readLong();
         shardId = new ShardId(in);
         globalCheckpoint = in.readZLong();
+        if (in.getVersion().onOrAfter(Version.V_8_0_0)) {
+            startingSeqNo = in.readZLong();
```

Comment:
I wonder if we should call this `trimAboveSeqNo`, similar to how we have it in `ResyncReplicationRequest`. It's clearer in that case what  the purpose is

Reply:
I started with `trimAboveSeqNo` then replaced with `startingSeqNo`. I pushed https://github.com/elastic/elasticsearch/pull/44756/commits/9582f1cb378433eee1b80a5fd4734ab75af7d8e4.